core/templating/templating.go

```diff
@@ -38,48 +48,89 @@ func NewTemplator() *Templator {
-		raymond.RegisterHelper("iso8601DateTimePlusDays", t.iso8601DateTimePlusDays)
-		raymond.RegisterHelper("currentDateTime", t.currentDateTime)
-		raymond.RegisterHelper("currentDateTimeAdd", t.currentDateTimeAdd)
-		raymond.RegisterHelper("currentDateTimeSubtract", t.currentDateTimeSubtract)
-		raymond.RegisterHelper("now", t.nowHelper)
-		raymond.RegisterHelper("randomString", t.randomString)
-		raymond.RegisterHelper("randomStringLength", t.randomStringLength)
-		raymond.RegisterHelper("randomBoolean", t.randomBoolean)
-		raymond.RegisterHelper("randomInteger", t.randomInteger)
-		raymond.RegisterHelper("randomIntegerRange", t.randomIntegerRange)
-		raymond.RegisterHelper("randomFloat", t.randomFloat)
-		raymond.RegisterHelper("randomFloatRange", t.randomFloatRange)
-		raymond.RegisterHelper("randomEmail", t.randomEmail)
-		raymond.RegisterHelper("randomIPv4", t.randomIPv4)
-		raymond.RegisterHelper("randomIPv6", t.randomIPv6)
-		raymond.RegisterHelper("randomUuid", t.randomUuid)
-		raymond.RegisterHelper("replace", t.replace)
-		raymond.RegisterHelper("faker", t.faker)
-
+		helperMethodMap["iso8601DateTime"] = t.iso8601DateTime
+		helperMethodMap["iso8601DateTimePlusDays"] = t.iso8601DateTimePlusDays
+		helperMethodMap["currentDateTime"] = t.currentDateTime
+		helperMethodMap["currentDateTimeAdd"] = t.currentDateTimeAdd
+		helperMethodMap["currentDateTimeSubtract"] = t.currentDateTimeSubtract
+		helperMethodMap["now"] = t.nowHelper
+		helperMethodMap["randomString"] = t.randomString
+		helperMethodMap["randomStringLength"] = t.randomStringLength
+		helperMethodMap["randomBoolean"] = t.randomBoolean
+		helperMethodMap["randomInteger"] = t.randomInteger
+		helperMethodMap["randomIntegerRange"] = t.randomIntegerRange
+		helperMethodMap["randomFloat"] = t.randomFloat
+		helperMethodMap["randomFloatRange"] = t.randomFloatRange
+		helperMethodMap["randomEmail"] = t.randomEmail
+		helperMethodMap["randomIPv4"] = t.randomIPv4
+		helperMethodMap["randomIPv6"] = t.randomIPv6
+		helperMethodMap["randomUuid"] = t.randomUuid
+		helperMethodMap["replace"] = t.replace
+		helperMethodMap["faker"] = t.faker
+		helperMethodMap["body"] = t.requestBody
+
+		raymond.RegisterHelpers(helperMethodMap)
 		helpersRegistered = true
 	}
 
-	return &Templator{}
+	return &Templator{
+		SupportedMethodMap: helperMethodMap,
+	}
+}
+
+func (t *Templator) SetLiterals(literals *models.Literals) {
+	literalMap := make(map[string]interface{})
+
+	if literals != nil {
+		for _, literal := range *literals {
+			literalMap[literal.Name] = literal.Value
+		}
+	}
+	t.literals = literalMap
+
+}
+func (t *Templator) SetRequestIndependentVariables(vars *models.Variables) error {
+
+	variableMap := make(map[string]interface{})
+	if vars != nil {
+		for _, variable := range *vars {
+			if variable.Function != REQUEST_BODY_HELPER {
+				value, error := t.callHelper(variable)
+				if error != nil {
+					return error
+				}
+				variableMap[variable.Name] = value
```

Comment:
this is going to execute the helper method once, and stored a constant value in the map. It's not gonna work if one is expecting random data or the current timestamp. I think we need to store the variables in the map and then execute the function only when we generate the context data for template rendering.

Reply:
done

Reply:
@tommysitu  incorporated all provided review comment changes

Reply:
@tommysitu  I have also handled concurrency while updating literals and variables. Added changes related to the same.

Reply:
have been busy, i will review your PRs today @kapishmalik

Reply:
np... Take ur time.